fwoa-tools/src/migrationVerify.ts

```diff
@@ -0,0 +1,140 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+import { readFileSync, WriteStream, createWriteStream } from 'fs';
+import { HealthLake, S3 } from 'aws-sdk';
+import { aws4Interceptor } from 'aws4-axios';
+import axios from 'axios';
+import * as dotenv from 'dotenv';
+import yargs from 'yargs';
+import { ExportOutput, getFhirClient, getFhirClientSMART, verifyResource } from './migrationUtils';
+
+dotenv.config({ path: '.env' });
+const { DATASTORE_ID, DATASTORE_ENDPOINT, API_AWS_REGION, IMPORT_OUTPUT_S3_BUCKET_NAME } = process.env;
+
+const IMPORT_VERIFICATION_LOG_FILE_PREFIX: string = 'import_verification_';
+
+// eslint-disable-next-line security/detect-non-literal-fs-filename
+const logs: WriteStream = createWriteStream(
+  `${IMPORT_VERIFICATION_LOG_FILE_PREFIX}${Date.now().toString()}.log`,
+  {
+    flags: 'a'
+  }
+);
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function parseCmdOptions(): any {
+  return yargs(process.argv.slice(2))
+    .usage('Usage: $0 [--smart, -s boolean] [--dryRun, -d boolean]')
+    .describe('smart', 'Whether the FWoA deployment is SMART-on-FHIR or not')
+    .boolean('smart')
+    .default('smart', false)
+    .alias('s', 'smart')
+    .describe('dryRun', 'Check operations and authentication status')
+    .boolean('dryRun')
+    .default('dryRun', false)
+    .alias('d', 'dryRun').argv;
+}
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const argv: any = parseCmdOptions();
+const smartClient: boolean = argv.smart;
+const dryRun: boolean = argv.dryRun;
+async function verifyFolderImport(): Promise<void> {
+  const outputFile: ExportOutput = JSON.parse(readFileSync('migrationExport_Output.txt').toString());
+  const fileNames = outputFile.file_names;
+
+  for (let k = 0; k < Object.keys(fileNames).length; k++) {
+    // eslint-disable-next-line security/detect-object-injection
+    const resourceType = Object.keys(fileNames)[k];
+    // eslint-disable-next-line security/detect-object-injection
+    const resourcePaths = fileNames[resourceType];
+    console.log(`Starting import for resource ${resourceType}`);
+    const interceptor = aws4Interceptor({
+      region: API_AWS_REGION!,
+      service: 'healthlake'
+    });
+
+    const fhirClient = await (smartClient ? getFhirClientSMART() : getFhirClient());
+    const s3Client = new S3({
```

Comment:
Can we test this under heavy load once (maybe something around 1GB)? I remember there being issues with the assumed credentials for the s3Client expiring part way through verifying a large number of resources, in which case we may need to create it again or change some other configuration. We may also want to ensure the FWoA access token doesn't expire, but that can be done through customer-side configuration as well as long as its documented.

Reply:
I can test this during the happy path testing today. We can either renew client or use connection keep-alive